spec/db.spec.ts

```diff
@@ -123,7 +123,30 @@ describe('db', () => {
           : describe;
 
         describeSSH('connect with ssh', () => {
-          const sshHost = process.env.SSH_HOST || 'localhost';
+          let serverInfo = {} as ServerConfig;
+
+          beforeEach(() => {
+            serverInfo = {
+              ...config[dbAdapter],
+              name: dbAdapter,
+              adapter: dbAdapter,
+              ssh: {
+                host: process.env.SSH_HOST || 'localhost',
+                port: 2222,
+                user: 'sqlectron',
+              },
+            };
+
+            // Must use the container name and the internal port
+            // in order to properly be able to connect the openssh-server container
+            // when the connection is being forward through the ssh session
+            serverInfo.host = dbAdapter;
+            if (dbAdapter === 'mariadb') {
+              serverInfo.port = 3306;
+            } else if (dbAdapter === 'redshift') {
+              serverInfo.port = 5432;
+            }
+          });
```

Comment:
why have this `beforeEach` here? Seems this would accomplish what we want and also name the eslint ignores below unnecessary?

```suggestion
          const serverInfo: ServerConfig = {
            ...config[dbAdapter],
            name: dbAdapter,
            adapter: dbAdapter,
            host: dbAdapter,
            ssh: {
              host: process.env.SSH_HOST || 'localhost',
              port: 2222,
              user: 'sqlectron',
            },
          };

          // Must use the container name and the internal port
          // in order to properly be able to connect the openssh-server container
          // when the connection is being forward through the ssh session
          if (dbAdapter === 'mariadb') {
            serverInfo.port = 3306;
          } else if (dbAdapter === 'redshift') {
            serverInfo.port = 5432;
          }
```

Reply:
I think we can move it out of beforeEach as you suggested. But the ignore will still be required, because it is being used for `ssh` attribute which can be undefined.

Reply:
Yeah, not sure if we will need it or not. Not sure how powerful tsc is in that regards for seeing that the `ssh` field is definitely defined for these tests.

Reply:
You were right, tsc is smart enough to detect that. I couldn't define the object only once at the describe scope because it would get mixed by each test, so I just moved it into a function.